Examples/Swift/ViewController.swift

```diff
@@ -51,61 +55,40 @@ class ViewController: UIViewController, MGLMapViewDelegate {
             return
         }
         
-        let coordinates = mapView.convert(sender.location(in: mapView), toCoordinateFrom: mapView)
+        if let annotation = mapView.annotations?.last, waypoints.count > 2 {
+            mapView.removeAnnotation(annotation)
+        }
         
-        howToBeginLabel.isHidden = true
-        getRouteButton.isHidden = false
-        clearMapButton.isHidden = false
+        if waypoints.count > 1 {
+            waypoints = Array(waypoints.suffix(1))
+        }
+        
+        let coordinates = mapView.convert(sender.location(in: mapView), toCoordinateFrom: mapView)
         
         let annotation = MGLPointAnnotation()
         annotation.coordinate = coordinates
         mapView.addAnnotation(annotation)
         
+        longPressHintView.isHidden = true
+        
         let waypoint = Waypoint(coordinate: coordinates)
         waypoint.coordinateAccuracy = -1
         waypoints.append(waypoint)
+        
+        requestRoute()
     }
     
-    @IBAction func didTapStartNavigation(_ sender: Any) {
-        startNavigation(along: userRoute!)
-    }
-    
-    @IBAction func didTapClearmap(_ sender: Any) {
-        mapView.removeAnnotations(mapView.annotations ?? [])
+    @IBAction func replay(_ sender: Any) {
+        let bundle = Bundle(for: ViewController.self)
```

Comment:
This class is inside the application target, so you can say simply `Bundle.main`. It’s only necessary to use `Bundle(for:)` inside a framework, since the framework isn’t the main bundle.